debezium-server/debezium-server-redis/src/test/java/io/debezium/server/redis/RedisStreamIT.java

```diff
@@ -61,15 +74,111 @@ void connectorCompleted(@Observes ConnectorCompletedEvent event) throws Exceptio
         }
     }
 
+    private PostgresConnection getPostgresConnection() {
+        return new PostgresConnection(Configuration.create()
+                .with("hostname", dbHostname)
+                .with("port", dbPort)
+                .with("user", dbUser)
+                .with("password", dbPassword)
+                .with("dbname", dbName)
+                .build());
+    }
+
     @Test
     public void testRedisStream() throws Exception {
-        Testing.Print.enable();
+        // Verifies that all the records of a PostgreSQL table are streamed to Redi
+        final int MESSAGE_COUNT = 4;
+        final String STREAM_NAME = "testc.inventory.customers";
+
         final List<StreamEntry> entries = new ArrayList<>();
         Awaitility.await().atMost(Duration.ofSeconds(RedisTestConfigSource.waitForSeconds())).until(() -> {
             final List<StreamEntry> response = jedis.xrange(STREAM_NAME, null, null, MESSAGE_COUNT);
+            entries.clear();
             entries.addAll(response);
             return entries.size() >= MESSAGE_COUNT;
         });
-        Assertions.assertThat(entries.size() >= MESSAGE_COUNT);
+        assertTrue("Redis Basic Stream Test Failed", entries.size() == MESSAGE_COUNT);
+    }
+
+    @Test
+    @FixFor("DBZ-4510")
+    public void testRedisConnectionRetry() throws Exception {
+        /*
+         * Test retry mechanism when encountering Redis connectivity issues:
+         * 1. Make Redis to be unavailable while the server is up
+         * 2. Create a new table named redis_test in PostgreSQL and insert 5 records to it
+         * 3. Bring Redis up again and make sure these records have been streamed successfully
+         */
```

Comment:
Should be a JavaDoc comment on the method itself.

Reply:
done